engine/access/rest/request/get_block.go

```diff
@@ -0,0 +1,116 @@
+package request
+
+import (
+	"fmt"
+
+	"github.com/onflow/flow-go/model/flow"
+)
+
+const heightQuery = "height"
+const startHeightQuery = "start_height"
+const endHeightQuery = "end_height"
+const MaxAllowedHeights = 50
+const idParam = "id"
+
+type GetBlock struct {
+	Heights      []uint64
+	StartHeight  uint64
+	EndHeight    uint64
+	FinalHeight  bool
+	SealedHeight bool
+}
+
+func (g *GetBlock) Build(r *Request) error {
+	return g.Parse(
+		r.GetQueryParams(heightQuery),
+		r.GetQueryParam(startHeightQuery),
+		r.GetQueryParam(endHeightQuery),
+	)
+}
+
+func (g *GetBlock) HasHeights() bool {
+	return len(g.Heights) > 0
+}
+
+func (g *GetBlock) Parse(rawHeights []string, rawStart string, rawEnd string) error {
+	var height Height
+	err := height.Parse(rawStart)
+	if err != nil {
+		return err
+	}
+	g.StartHeight = height.Flow()
+	err = height.Parse(rawEnd)
+	if err != nil {
+		return err
+	}
+	g.EndHeight = height.Flow()
+
+	var heights Heights
+	err = heights.Parse(rawHeights)
+	if err != nil {
+		return err
+	}
+	g.Heights = heights.Flow()
+
+	// if both height and one or both of start and end height are provided
+	if len(g.Heights) > 0 && (g.StartHeight != EmptyHeight || g.EndHeight != EmptyHeight) {
+		return fmt.Errorf("can only provide either heights or start and end height range")
+	}
+
+	// if neither height nor start and end height are provided
+	if len(heights) == 0 && (g.StartHeight == EmptyHeight || g.EndHeight == EmptyHeight) {
+		return fmt.Errorf("must provide either heights or start and end height range")
+	}
+
+	if g.StartHeight > g.EndHeight {
+		return fmt.Errorf("start height must be less than or equal to end height")
+	}
+	// check if range exceeds maximum but only if end is not equal to special value which is not known yet
+	if g.EndHeight-g.StartHeight > MaxAllowedHeights && g.EndHeight != FinalHeight && g.EndHeight != SealedHeight {
```

Comment:
heights are inclusive right? Should this be 
```
g.EndHeight-g.StartHeight+1 > MaxAllowedHeights
```

Reply:
yep, or `>=`